modules/openxr/openxr_api.cpp

```diff
@@ -513,40 +513,66 @@ void OpenXRAPI::copy_string_to_char_buffer(const String p_string, char *p_buffer
 	}
 }
 
+PackedStringArray OpenXRAPI::get_all_requested_extensions() {
+	// This returns all extensions we will request regardless of whether they are available.
+	// This is mostly used by the editor to filter features not enabled through project settings.
+
+	PackedStringArray requested_extensions;
+	for (OpenXRExtensionWrapper *wrapper : registered_extension_wrappers) {
+		const HashMap<String, bool *> &wrapper_request_extensions = wrapper->get_requested_extensions();
+
+		for (const KeyValue<String, bool *> &requested_extension : wrapper_request_extensions) {
+			if (!requested_extensions.has(requested_extension.key)) {
+				requested_extensions.push_back(requested_extension.key);
+			}
+		}
+	}
+
+	return requested_extensions;
+}
+
 bool OpenXRAPI::create_instance() {
 	// Create our OpenXR instance, this will query any registered extension wrappers for extensions we need to enable.
 
-	// Append the extensions requested by the registered extension wrappers.
-	HashMap<String, bool *> requested_extensions;
+	// We can request an extension multiple times if there are dependencies
+	struct request_extension {
```

Comment:
Even though this is a very local struct, I think we always use camel case for struct names - using snake case feels a little weird